### PR TITLE
BlockWorlds Entities

### PR DESCRIPTION
**Pull request by MikiGamer**

### CODE REVIEW

File: datasrc/network.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 	"BUG_DDRACE_INPUT", "BUG_FNG_LASER_RANGE", "BUG_VANILLA_BOUNCE",
 	"PREDICT_FNG", "PREDICT_DDRACE", "PREDICT_DDRACE_TILES", "PREDICT_VANILLA",
 	"ENTITIES_DDNET", "ENTITIES_DDRACE", "ENTITIES_RACE", "ENTITIES_FNG",
-	"ENTITIES_VANILLA", "DONT_MASK_ENTITIES",
+	"ENTITIES_VANILLA", "ENTITIES_BW", "DONT_MASK_ENTITIES",
```
Comment (heinrich5991) [inline datasrc/network.py:20]: This needs to go at the end.

File: src/game/client/components/mapimages.cpp [hunk 2]
```diff
@@ -108,6 +108,8 @@ IGraphics::CTextureHandle CMapImages::GetEntities()
 		pEntities = "fng";
 	else if(GameClient()->m_GameInfo.m_EntitiesVanilla)
 		pEntities = "vanilla";
+	else if(GameClient()->m_GameInfo.m_EntitiesBW)
+		pEntities = "blockworlds";
```
Comment (heinrich5991) [inline src/game/client/components/mapimages.cpp:112]: Can you move this up over FNG?

File: src/game/client/gameclient.cpp [hunk 3]
```diff
@@ -1065,6 +1066,7 @@ static CGameInfo GetGameInfo(const CNetObj_GameInfoEx *pInfoEx, int InfoExSize,
 		Info.m_EntitiesRace = Flags&GAMEINFOFLAG_ENTITIES_RACE;
 		Info.m_EntitiesFNG = Flags&GAMEINFOFLAG_ENTITIES_FNG;
 		Info.m_EntitiesVanilla = Flags&GAMEINFOFLAG_ENTITIES_VANILLA;
+		Info.m_EntitiesBW = Flags&GAMEINFOFLAG_ENTITIES_BW;
```
Comment (heinrich5991) [inline src/game/client/gameclient.cpp:1069]: You need to create a new version for this, and put it into an `if` block:
```
if(Version >= 4)
{
	Info.m_EntitiesBW = Flags&GAMEINFOFLAG_ENTITIES_BW;
}
```

The server also needs to be updated to send version 4.

**Review by heinrich5991 — CHANGES_REQUESTED** [review]
Thanks for your first pull request. :)

It still needs some changes to maintain backward compatibility. Ping me on Discord (heinrich5991#9827) if you need help with the comments I left.

**Comment by heinrich5991** [comment]
Oh, sorry, missed the update.

You still need to update the version number the server sends in its `CNetObj_GameInfoEx`.

**Review by heinrich5991 — APPROVED** [review]
Thanks a lot. :)

bors r+

**Comment by fokkonaut** [comment]
Why did you merge this? The entities_clear are not good for ingame playing, and BW has multiple layers (you can add unlimited front layer like layers). This should be reverted

**Comment by fokkonaut** [comment]
@heinrich5991

**Comment by heinrich5991** [comment]
> Why did you merge this?

In case this was not a rhetorical question: Because having entities seemed like a better option than not having them.

**Comment by fokkonaut** [comment]
But they are not working properly because as I said BW has multiple front layers

**Pull request merged**